### PR TITLE
docs: add Mac M1 build conan

### PR DESCRIPTION
**Pull request by AnthonyLaw**

### CODE REVIEW

File: docs/BUILD-conan.md [hunk 1]
```diff
@@ -45,16 +45,49 @@ Following instructions should work on Mac, Linux (Ubuntu 20.04) and Windows.
...
+     ```
+
+  3. Install [Conan](https://conan.io/downloads.html).
+
+  4. Set the right config for conan:
+
+      ```sh
+      conan profile new default --detect
+      conan config set general.revisions_enabled=True
+      ```
```
Comment (segfaultxavi) [inline docs/BUILD-conan.md:69]: For simplicity, I would merge this new section with the previous one:
- Add `pkg-config` to the `brew install` line (it does not hurt to always install it, right?)
- Add step 4 as it is, with a note:
  ```
  4. Set the right config for Conan (Only for Mac M1):
  ```

File: docs/BUILD-conan.md [hunk 2]
```diff
@@ -45,16 +45,49 @@ Following instructions should work on Mac, Linux (Ubuntu 20.04) and Windows.
...
+
+**NOTE: Only for Mac (M1)**
+
+Minor change required in `conanfile.txt`.
+
+- Change `boost:without_math=True`.
+- Add `rocksdb:enable_sse = False` under `rocksdb:shared = True`.
+
+___
+
```
Comment (segfaultxavi) [inline docs/BUILD-conan.md:82]: I would turn this note into step 5 in the above list:
```
5. Modify `conanfile.txt` (Only for Mac M1):
```

File: docs/BUILD-conan.md [hunk 3]
```diff
@@ -45,16 +45,49 @@ Following instructions should work on Mac, Linux (Ubuntu 20.04) and Windows.
...
 as this will probably take *a bit*.
 
 ```sh
-conan remote add nemtech https://catapult.jfrog.io/artifactory/api/conan/ngl-conan
+conan remote add symbol https://catapult.jfrog.io/artifactory/api/conan/ngl-conan
 
-git clone https://github.com/nemtech/catapult-server.git
-cd catapult-server
+git clone https://github.com/symbol/catapult-client.git
+cd catapult-client
```
Comment (segfaultxavi) [inline docs/BUILD-conan.md:90]: This should go to a different commit, it's unrelated to the M1 instructions.

**Review by segfaultxavi — CHANGES_REQUESTED** [review]
I prefer to keep machine-specific instructions to a minimum, otherwise, this will explode in the future...

**Comment by AnthonyLaw** [comment]
> I prefer to keep machine-specific instructions to a minimum, otherwise, this will explode in the future...

Agree, i made some change for that ;)

**Review by segfaultxavi — APPROVED** [review]
Sorry for the late review, I was on holiday!

File: docs/BUILD-conan.md [hunk 4]
```diff
@@ -40,11 +40,22 @@ Following instructions should work on Mac, Linux (Ubuntu 20.04) and Windows.
   2. Install build dependencies:
 
      ```sh
-     brew install git cmake ninja
+     brew install git cmake ninja pkg-config
      ```
 
   3. Install [Conan](https://conan.io/downloads.html).
 
+  4. Set the right config for Conan (Only for Mac M1):
```
Comment (Jaguar0625) [inline docs/BUILD-conan.md:48]: can you please remove `(Only for Mac M1)` from 4?

**Review by segfaultxavi — APPROVED** [review]

**Comment by Jaguar0625** [comment]
Merged in d4e11e01bceefa1ffcf854f34f168b00e42127e2.

**Pull request closed without merging**